portfolio/src/main/java/com/google/sps/utils/CommentDataStore.java

```diff
@@ -30,21 +27,29 @@ public void store(Comment comment) {
 
     /**
      * Loads the comments from Datastore.
-     * @return An Iterable.
+     * @return An immutable list of comments.
      */
     @Override
-    public List<Comment> load() {
-
+    public ImmutableList<Comment> load(int limit) {
         Query query = new Query(KIND).addSort(TIMESTAMP, Query.SortDirection.DESCENDING);
-
         PreparedQuery results = this.datastoreService.prepare(query);
+        Iterable<Entity> resultsIterable = results.asIterable(FetchOptions.Builder.withLimit(limit));
 
-        List<Comment> comments = new ArrayList<>();
-        for (Entity entity: results.asIterable()) {
-            Comment comment = Comment.CREATOR.fromEntity(entity);
-            comments.add(comment);
+        ImmutableList.Builder<Comment> comments = ImmutableList.builder();
+        for( Entity entity: resultsIterable){
+            comments.add(Comment.CREATOR.fromEntity(entity));
         }
+        return comments.build();
+    }
 
-        return comments;
+    /**
+     * Deletes a specific comment in Datastore.
+     * @param kind Comment entity kind name.
+     * @param id Target comment entity id.
+     */
+    @Override
+    public void delete(String kind, long id) {
+        Key taskEntityKey = KeyFactory.createKey(kind, id);
+        this.datastoreService.delete(taskEntityKey);
```

Comment:
Do we need to check existence first or it's handled by datastoreService?

Reply:
Deleting a nonexistent entity has no effect, will not cause error

Reply:
I will add a TODO to handle "java.lang.IllegalArgumentException - If the specified key was invalid."